components/sync_manager/src/syncmanager.udl

```diff
@@ -0,0 +1,106 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+namespace syncmanager { };
+
+[Error]
+enum SyncManagerError {
+    "UnknownEngine",
+    "UnsupportedFeature",
+    "Sync15Error",
+    "UrlParseError",
+    "InterruptedError",
+    "JsonError",
+    "LoginsError",
+    "PlacesError",
+    "AnyhowError",
+};
+
+dictionary SyncParams {
+    // Why are we performing this sync?
+    SyncReason reason;
+    // Which engines should we sync?
+    SyncEngineSelection engines;
+    // TODO
```

Comment:
The changes that this sync should make to the "account global" list of enabled/disabled engines (for example, if the UI was used to change an engine's state since the last sync.